### PR TITLE
Castling

### PR DESCRIPTION
**Pull request by saakuffo**
Castling has been implemented and works however it is using stub codes for logic it depends on
**Side note, the king and queen had their starting locations flipped in initialization. I flipped them back.**

- Because we will have the specific king we are referring to (what the player clicks on/moves) and the specific rook (what the player clicks on/moves over), I am passing in a specific king and a specific rook
(just to confirm, I am assuming and they do in fact seem to be, passed by reference)

- after making sure the piece is not obstructed, it checks to make sure the pieces can castle (this will be changed when the functionality is implemented

- during the check to see if it can castle, it checks to make sure this is the first move. If it is, castling can proceed.

**More logic is going to be needed for first_move. is_obstructed is already in the works**
this code will be changed to remove the test parameters to allow me to test all lines of logic written.

### CODE REVIEW

File: app/models/game.rb [hunk 1]
```diff
@@ -69,4 +69,27 @@ def is_game_in_check?(opponents_color)
...
+  def castle(king, rook, test_fm=true, test_iso=false) # STUB CODE NEEDS TO BE CHANGED WHEN IS OBSTRUCTED/FIRST MOVE IS DONE
+    current_king = king
+    current_rook = rook
+    if can_castle?(current_king, current_rook, test_fm, test_iso) # STUB CODE NEEDS TO BE CHANGED WHEN IS OBSTRUCTED/FIRST MOVE IS DONE
+      if rook.x_position < king.x_position
+        current_king.x_position = current_king.x_position - 2
+        current_rook.x_position = current_rook.x_position + 2
+      else
+        current_king.x_position = current_king.x_position + 2
+        current_rook.x_position = current_rook.x_position - 3
```
Comment (lasley) [inline app/models/game.rb]: 3?

Comment (saakuffo) [inline app/models/game.rb]: Yup 3. Queen side Castling is a 3 move on the Rook. Watched a really old man who looked good at chess correct my belief.

Comment (lasley) [inline app/models/game.rb]: 😆  cool, just making sure -looked like a typo

**Comment by saakuffo** [comment]
Some exploded when I tried to merge in master. I think I merged when I should have re based or whatevs.
Went commit diving and fixed the issue.  Most of the files that look 'changed' are not changed at all (same number of deletions and additions) and someone them are changed from the merge.

File: .byebug_history [hunk 2]
```diff
@@ -0,0 +1,34 @@
+exit
```
Comment (lasley) [inline .byebug_history]: This file should not be checked in and should be in gitignore

**Pull request merged**